tracer/src/Datadog.Trace/Logging/DirectSubmission/DirectLogSubmissionSettings.cs

```diff
@@ -97,9 +97,54 @@ public DirectLogSubmissionSettings(IConfigurationSource? source, IConfigurationT
 
             ApiKey = config.WithKeys(ConfigurationKeys.ApiKey).AsRedactedString();
 
-            LogsInjectionEnabled = config.WithKeys(ConfigurationKeys.LogsInjectionEnabled).AsBool();
+            var isEnabled = DirectLogSubmissionEnabledIntegrations.Count > 0;
```

Comment:
I'm not sure if there's much point in having `DirectLogSubmissionSettings` _and_ `ImmutableDirectLogSubmissionSettings` after these changes? IIRC the original split was to make it easier to test these things in isolation, but if that's no longer the case, it's probably just needless copying and allocation?